zulip_bots/zulip_bots/bots/susi/susi.py

```diff
@@ -41,9 +41,39 @@ def handle_message(self, message: Dict[str, str], bot_handler: Any) -> None:
             return
         reply = requests.get("https://api.susi.ai/susi/chat.json", params=dict(q=msg))
         try:
-            answer = reply.json()['answers'][0]['actions'][0]['expression']
-        except Exception:
+            data = reply.json()
+            actions = data['answers'][0]['actions']
+            answer = ""
+            for action in actions:
+                if action['type'] == 'answer':
+                    answer += action['expression'] + '\n'
+                elif action['type'] == 'anchor':
+                    text = action['text']
+                    link = action['link']
+                    answer += '[' + text + '](' + link+')' + '\n'
+                elif action['type'] == 'table':
+                    tableData = data['answers'][0]['data']
+                    msg = ""
+                    columns = list(action['columns'].keys())
+                    columnsData = list(action['columns'].values())
+                    # using markdown make a table
+                    for heading in columnsData:
+                        answer += "| " + heading + " "
+                    answer += "\n"
+                    for heading in columns:
+                        answer += "|" + "------" + " "
+                    answer += "\n"
+                    for item in (tableData):
+                        if (item[columns[0]]):
+                            column1 = item[columns[0]] if (type(item[columns[0]]) is str) else item[columns[0]][0]
+                            column2 = item[columns[1]] if (type(item[columns[1]]) is str) else item[columns[1]][0]
+                            column3 = item[columns[2]] if (type(item[columns[2]]) is str) else item[columns[2]][0]
+                            msg = "| " + column1+" | " + column2 + " | " + column3 + "| "
+                            answer = answer + msg + "\n"
+                    answer += "\n"
+        except Exception as e:
```

Comment:
too broad exception here. What exceptions do you think would arise ?

Reply:
@tonythomas01 there can be many exceptions. Sometimes, the susi server might not reply with the expected object. This may result in exceptions as we are processing assuming a particular response object.